DOCS-2675: Document cloud inference (#4322)
viamrobotics/docs: docs/data-ai/ai/run-inference.md

description: "Run inference on a model with a vision service or an SDK."
---

Inference is the process of generating output from a machine learning (ML) model.
With Viam, you can run inference to generate the following kinds of output:

- object detection (using bounding boxes)
- classification (using tags)

You can run inference locally on a Viam machine, or remotely in the Viam cloud.

> ## Machine

Reviewer: or Machine Inference maybe

You can use `viam-server` to deploy and run ML models directly on your machines.

You can run inference on your machine in the following ways:

- with a vision service
- manually in application logic with an SDK

> Typical machines, like Raspberry Pi 4, can run small TensorFlow Lite (TFLite) models.

Reviewer: I don't know if we want to call them "typical" or something that describes why they can / can't run certain models based on their spec

Author: Agreed. Rephrased both sentences and changed this to 'entry-level', which is still not strictly 100% perfectly true, but I think is near enough.

{{% /tab %}}
{{< /tabs >}}

{{% /tab %}}
{{< /tabs >}}

## Cloud inference

> Cloud inference enables you to run machine learning models in the Viam cloud, instead of on a local machine.
> Cloud inference runs on powerful servers with more computing power than most edge devices.
> Because cloud inference has access to more computing power, you can benefit from:

Author: ```suggestion
Cloud inference enables you to run machine learning models in the Viam cloud, instead of on a local machine.
Cloud inference often provides more computing power than edge devices, so you can benefit from:
```

- larger, more accurate models
- faster inference times

You can run cloud inference using any Tensorflow model in the Viam registry, including private models owned by or shared with your organization.

To run cloud inference, you must pass

- the binary data ID and organization of the data you want to run inference on
- the name, version, and organization of the model you want to use for inference

> The [`viam infer`](/dev/tools/cli/#infer) CLI command runs inference in the cloud on a piece of data using the specified ML model:

Author: Unfortunately not! I originally thought there was a way but only the protos exist right now. That's also why I wound up messing with the SDK doc generation scripts (now reverted), because I was getting some typescript failures and I had to fix them to continue investigating.
